photon-client/src/integTest/scala/com/linkedin/photon/ml/estimators/GameEstimatorTest.scala

```diff
@@ -14,28 +14,212 @@
+          "updating-sequence" -> "global",
+          "normalization-type" -> normalizationType.toString,
+          "train-input-dirs" -> "",
+          "validate-input-dirs" -> "",
+          "output-dir" -> "",
+          "feature-name-and-term-set-path" -> ""
+        )))
+    })
+
+  /**
+   * In this test, we just verify that we can fit a model with each possible normalization type, without verifying that
+   * the results are correct.
+   *
+   * Not using DataProviders, because they can't be composed to have multiple DataProviders serving arguments to
+   * a single function.
+   */
+  @Test
+  def testNormalization(): Unit = sparkTest("testNormalization") {
+
+    for (params: GameParams <- normalizationParameters()(0)) {
+
+      // This example has only a single fixed effect
+      val (coordinateId, featureShardId) = ("global", "features")
+      val labeledPoints: Seq[LabeledPoint] = trivialLabeledPoints()(0)(0)
+      val nDimensions = labeledPoints.head.features.size
+
+      // Setup feature names a feature index from feature name to feature index
+      // Have to drop in an intercept "feature" but intercepts are turned ON by default
+      val featureNames = (0 until nDimensions).map(i => s"feature-$i").toSet
+      val featureIndexMap = DefaultIndexMap(featureNames) + ((GLMSuite.INTERCEPT_NAME_TERM, nDimensions))
+
+      // Generate a Spark DataFrame containing labeled points (label, x, y)
+      // :+ 1.0 is because we need to add that column to make up a correct design matrix, including that 1.0
+      // to multiply by the intercept term.
+      val label = GameConverters.FieldNames.RESPONSE
+      val trainingData: DataFrame = new SQLContext(sc)
+        .createDataFrame(labeledPoints
+          .map { point: LabeledPoint =>
+            val x = Vectors.dense(point.features.toDenseVector.toArray :+ 1.0).toSparse
+            (point.label, x)
+          })
+        .toDF(label, featureShardId)
+
+      val stats = BasicStatisticalSummary(trainingData.select(featureShardId).map(_.getAs[SparkVector](0)))
+
+
+      // Compute normalization contexts based on statistics of the training data for this (unique) feature shard
+      val normalizationContexts: Option[Map[String, NormalizationContext]] =
+        Some(Map((featureShardId, stats))
+          .mapValues { featureShardStats =>
+            val intercept: Option[Int] = featureIndexMap.get(GLMSuite.INTERCEPT_NAME_TERM)
+            NormalizationContext(params.normalizationType, featureShardStats, intercept)
+          })
+
+      // Create GAMEEstimator and fit model
+      // Returns (model, evaluation, optimizer config)
+      val models: Seq[(GAMEModel, Option[EvaluationResults], String)] =
+      createEstimator(params, "simpleTest")._1.fit(trainingData, validationData = None, normalizationContexts)
+
+      val model = models.head._1.getModel(coordinateId).head.asInstanceOf[FixedEffectModel].model
+      for (i <- 0 until 3)
```

Comment:
We want to use them - but here I would need two, which is not supported (and this is explained in the comment).